### PR TITLE
Jworthington/t 2213 implement existing api spec in the

### PR DESCRIPTION
**Pull request by JWorthington97**
- removed admin / internal endpoints
- removed extraneous tags
- renamed all instances of Tillit to Two (except for Tillit AS)
- ordered operations alphabetically (through Redocly)
- created basic API landing page for Checkout API and Search API

The code blocks on the right are not properly formatted. Other PR is the fix for this

### CODE REVIEW

File: openapi/checkout-api.yaml [hunk 1]
```diff
@@ -0,0 +1,1651 @@
...
+paths:
+  '/buyers/{id}/addresses':
+    parameters:
+      - in: path
+        name: id
+        schema:
+          type: string
+        required: true
+        description: ID of the buyer
+    post:
```
Comment (jkrgr) [inline openapi/checkout-api.yaml:28]: only get address should be in the spec

Comment (jkrgr) [reply openapi/checkout-api.yaml:28]: And that is the GET address with organization number (not with buyer id, and not with address id)

File: openapi/checkout-api.yaml [hunk 2]
```diff
@@ -0,0 +1,1651 @@
...
+      tags:
+        - Address
+  '/invoice/{order_id}/pdf':
+    parameters:
+      - in: path
+        name: order_id
+        schema:
+          type: string
+        required: true
+        description: ID of the order
```
Comment (jkrgr) [inline openapi/checkout-api.yaml:133]: include the `lang` query parameter here for localization

File: openapi/checkout-api.yaml [hunk 3]
```diff
@@ -0,0 +1,1651 @@
...
+      summary: Generate a pdf invoice for an order.
+      responses:
+        '200':
+          description: OK
+          content:
+            application/pdf:
+              schema:
+                type: string
+                format: binary
+      description: Endpoint for creating a pdf invoice.
```
Comment (jkrgr) [inline openapi/checkout-api.yaml:144]: > Endpoint for downloading the invoice of a fulfilled order

File: openapi/checkout-api.yaml [hunk 4]
```diff
@@ -0,0 +1,1651 @@
...
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/FulfilledOrder'
+      description: |-
+        Set the order status to `FULFILLED`.
+
+        Fulfilled requests should normally have empty request body.
```
Comment (jkrgr) [inline openapi/checkout-api.yaml:213]: > Set the order status to `FULFILLED`(similar to "capture" for cards). The invoice will be issued and distributed it to the customer.

File: openapi/checkout-api.yaml [hunk 5]
```diff
@@ -0,0 +1,1651 @@
...
+                $ref: '#/components/schemas/RefundResponse'
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/RefundRequest'
+      description: Endpoint for refunding an order.
+      tags:
+        - Order
+  /order_intent:
```
Comment (jkrgr) [inline openapi/checkout-api.yaml:276]: Could you go over the examples and update them to be more realistic?

File: openapi/checkout-api.yaml [hunk 6]
```diff
@@ -0,0 +1,1651 @@
...
+                        phone_number: '+440123456789'
+                        email: john.doe@example.com
+                      company:
+                        country_prefix: 'NO'
+                        organization_number: '900000006'
+                        company_name: Acme Inc
+                    approved: true
+                    decline_reason: 'null'
+                    merchant_id: a42e70d0-2f1a-4c94-8ccb-1932ea933583
+                    tracking_id: 18dc34e0-4045-4951-a00e-146856feacc6
```
Comment (jkrgr) [inline openapi/checkout-api.yaml:330]: `tracking_id` should be removed. If the response schema returns this we should remove it from the schema as well

File: openapi/checkout-api.yaml [hunk 7]
```diff
@@ -0,0 +1,1651 @@
...
+                        phone_number: '+440123456789'
+                        email: john.doe@example.com
+                      company:
+                        country_prefix: 'NO'
+                        organization_number: '900000006'
+                        company_name: Acme Inc
+                    approved: false
+                    decline_reason: TOO_HIGH_RISK
+                    merchant_id: a42e70d0-2f1a-4c94-8ccb-1932ea933583
+                    tracking_id: null
```
Comment (jkrgr) [inline openapi/checkout-api.yaml:373]: same here

File: openapi/checkout-api.yaml [hunk 8]
```diff
@@ -0,0 +1,1651 @@
...
+                      email: john.doe@example.com
+                    company:
+                      country_prefix: 'NO'
+                      organization_number: '900000009'
+                      company_name: Acme Inc
+                  merchant_id: a42e70d0-2f1a-4c94-8ccb-1932ea933583
+      tags:
+        - Order intent
+        # - Order
+  '/order/{id}/verification':
```
Comment (jkrgr) [inline openapi/checkout-api.yaml:430]: internal endpoint, remove

File: openapi/checkout-api.yaml [hunk 9]
```diff
@@ -0,0 +1,1651 @@
...
+                  verification_method:
+                    type: string
+                  verified_on:
+                    type: string
+                    format: date-time
+                required:
+                  - verification_status
+      tags:
+        - Order
+  '/orders/{id}/external_order_status':
```
Comment (jkrgr) [inline openapi/checkout-api.yaml:463]: private endpoint

File: openapi/search-api.yaml [hunk 10]
```diff
@@ -0,0 +1,199 @@
...
+  version: '1.0'
+  contact:
+    name: Tillit AS
+    url: 'https://two.inc'
+    email: tech@two.inc
+  description: Two's Search API is used to improve the user experience in the merchant's checkout.
+servers:
+  - url: 'http://localhost:3000'
+paths:
+  '/search/company/address/{organization_number}':
```
Comment (jkrgr) [inline openapi/search-api.yaml:13]: yes, this is the only address endpoint we expose. then you can remove all the address endpoints in the other file

File: openapi/search-api.yaml [hunk 11]
```diff
@@ -0,0 +1,199 @@
...
+info:
+  title: Two Search API
+  version: '1.0'
+  contact:
+    name: Tillit AS
+    url: 'https://two.inc'
+    email: tech@two.inc
+  description: Two's Search API is used to improve the user experience in the merchant's checkout.
+servers:
+  - url: 'http://localhost:3000'
```
Comment (jkrgr) [inline openapi/search-api.yaml:11]: Let's use the actual urls. Example requests:
Norway: https://no.search.two.inc/search?limit=50&offset=0&q=mono
UK: : https://gb.search.two.inc/search?limit=50&offset=0&q=mono

So our base urls differ between the countries. Not sure how to represent that in the open api spec, but should be a way

File: openapi/search-api.yaml [hunk 12]
```diff
@@ -0,0 +1,199 @@
...
+        Clients can implement pagination by calling with same query q and different offset value. The limit parameter does not have to be the same in each request while doing pagination.
+
+
+        Base url:
+
+        `https://search-api-demo-j6whfmualq-lz.a.run.app/`
+
+        Example: 
+
+        `GET https://search-api-demo-j6whfmualq-lz.a.run.app/search?limit=50&offset=0&q=monoz`
```
Comment (jkrgr) [inline openapi/search-api.yaml:157]: update the examples to use the proper urls

**Review by jkrgr — CHANGES_REQUESTED** [review]

File: openapi/checkout-api.yaml (hunk 6 above)
Comment (JWorthington97) [reply openapi/checkout-api.yaml:330]: Removed from schema as well as here

File: openapi/checkout-api.yaml (hunk 2 above)
Comment (JWorthington97) [reply openapi/checkout-api.yaml:133]: @jkrgr I don't believe this is present in the checkout-api endpoint at the moment. Does it still need to be added?

**Comment by JWorthington97** [comment]
Also added authentication headers to endpoints that require it

**Comment by JWorthington97** [comment]
@jkrgr `Endpoint for accessing all orders made by a given merchant or buyer.` 

Is the get list of orders endpoint internal too? In checkout-api it requires AdminBasicAuth

**Comment by JWorthington97** [comment]
Updated all request / response examples with realistic data

File: openapi/checkout-api.yaml (hunk 2 above)
Comment (jkrgr) [reply openapi/checkout-api.yaml:27]: It's present, and yes it should be included: check `invoice_pdf_handler`,  `?lang=en_US` and `?lang=nb_NO`. You also have `?v=original` to download the original invoice and not the credit note (if any)

**Comment by jkrgr** [comment]
> @jkrgr Endpoint for accessing all orders made by a given merchant or buyer.
>Is the get list of orders endpoint internal too? In checkout-api it requires AdminBasicAuth

get orders by merchant or buyer (`retrieve_order_list_handler`) is `MerchantXApiKeyAuth` _or_ `AdminBasicAuth`. Note that url is `/v2/orders`. `retrieve_order_list_handler_legacy` should not be included.

**Review by jkrgr — APPROVED** [review]

**Pull request merged**